site/docs/v1/tech/lambdas/index.adoc

```diff
@@ -45,11 +54,98 @@ function populate(jwt, user, registration) {
 }
 ----
 
-Each lambda documentation page will have an example lambda implementation.
+=== Making an HTTP Request
+
+include::docs/v1/tech/shared/_advanced-edition-blurb.adoc[]
+
+Here is an example of a FusionAuth lambda that adds additional claims to a JWT based on an HTTP request:
+
+[source,javascript]
+----
+function populate(jwt, user, registration) {
+  var response = fetch("https://api.example.com/api/status?"+user.id, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  
+  if (response.status === 200) {
+    var jsonResponse = JSON.parse(response.body);
+    jwt.status = jsonResponse.status;
+  } else {
+    jwt.status = "basic";
+  }
+----
+
+==== Headers
+
+You can provider request header values in a number of different ways:
+
+[source,javascript,title="An anonymous object"]
+----
+headers: {
+  "Content-Type": "application/json"
+}
+----
+
+[source,javascript,title="A hash or map"]
+----
+headers: new Headers({
+   "Content-Type": "application/json"
+})
+----
+
+[source,javascript,title="An array"]
+----
+headers: new Headers([
+    ["Content-Type", "application/json"]
+])
+----
+
+==== Response 
+
+A response object will be returned. It will have the following fields:
+
+[.api]
+[field]#headers# [type]#[Array]#::
+The headers returned by the response.
+
+[field]#status# [type]#[Integer]#::
+The HTTP status code.
+
+[field]#body# [type]#[String]#::
+The body of the response.
 
 == JavaScript
 
-FusionAuth is using the Nashorn JavaScript engine to compile and execute lambdas functions. The Nashorn engine supports ECMA script version 5.1.
+=== Engine
+
+As of FusionAuth 1.35, you have the choice of JavaScript engine:
+
+* Nashorn
+* GraalJS
+
+Versions previous to 1.35 only have access to the Nashorn engine.
+
+==== Nashorn
+
+Nashorn is built on top of the Java virtual machine and while Nashorn permits access to the Java API, for security reasons FusionAuth restricts access to all Java objects during a lambda invocation. Here is the documentation provided by Oracle for the Nashorn engine:
+
+https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/nashorn/
+
+The Nashorn engine supports ECMAscript version 5.1.
+
+==== GraalJS
+
+GraalJS is built on top of the Java virtual machine. For security reasons, FusionAuth restricts access to various GraalJS features during a lambda invocation.
+
+Here is documentation for the GraalJS engine:
+
+https://github.com/oracle/graaljs
+
+The GraalJS Engine https://www.graalvm.org/22.0/reference-manual/js/JavaScriptCompatibility/[supports EcmaScript 2021].
+However, there is an https://github.com/FusionAuth/fusionauth-issues/issues/1630[open bug which interferes with this support] in version 1.35.
```

Comment:
Will we remember to remove this comment once this is resolved? It may be easier not to mention it since there is no regression from Nashorn, it is only a feature that should be available but is not yet.

Reply:
I can add a note to update this doc to that bug. I really want to let folks know that they can use more modern JS with graalvm, which is a reason to choose it even if you don't have the http request functionality.